app/scripts/acs/acs-service.js

```diff
@@ -68,14 +69,38 @@
                 tract: cols.tract,
                 srid: 4326
             });
-            return Util.makeRequest(cartodbsql, query).then(function (result) {
+
+            return Util.makeRequest(cartodbsql, query)
+            .then(function (result) {
                 var tractGroups = _(result).groupBy(function (tract) { // group by state+county
                     return 'state:' + tract.state + '+county:' + tract.county;
                 }).mapValues(function (tracts) { // tract objects to comma delimited values
                     return _.map(tracts, function (tract) { return tract.tract; }).join(',');
                 }).value();
-                $log.debug(tractGroups);
-                // TODO: query census api and return data
+
+                var acsQueries = [];
+                var acsQueries = _.map(tractGroups, function (value, key) {
+                    return $http.get(acsUrl, {
+                        cache: true,
+                        params: {
+                            get: _.keys(ACSVariables).join(','), 
+                            for: 'tract:' + value,
+                            in: key
+                        }
+                    });
+                });
+                return $q.all(acsQueries);
+            // Transform data into json object with all data from multiple results aggregated 
+            //  into a single array
+            }).then(function (results) {
+                var allData = _.reduce(results, function (total, n) {
+                    return total.concat(n.data.slice(1));
+                }, []);
+                var data = {
```

Comment:
At some point we will probably want another transformation of this data into objects so that the value of ACS variables can be looked up easily.